src/persistence/db/upgrades/dbto11.cpp

```diff
@@ -0,0 +1,290 @@
+    // since each group is a chat so is in peers, but authors no messages, and each group
+    // member is an author so is in peers, but has no chat.
+    // Splitting peers makes the relation much clearer and the tables have a single meaning.
+    QVector<RawDatabase::Query> upgradeQueries;
+    PeersToAuthors::peersToAuthors(db, upgradeQueries);
+    PeersToChats::peersToChats(db, upgradeQueries);
+    dropPeers(upgradeQueries);
+    upgradeQueries += RawDatabase::Query(QStringLiteral("PRAGMA user_version = 11;"));
+    bool transactionPass = db.execNow(upgradeQueries);
+    if (transactionPass) {
+        return db.execNow("VACUUM");
+    }
+    return transactionPass;
+}
+
+bool DbTo11::PeersToAuthors::peersToAuthors(RawDatabase& db, QVector<RawDatabase::Query>& upgradeQueries)
+{
+    createNewTables(upgradeQueries);
+    if (!populateAuthors(db, upgradeQueries)) {
+        return false;
+    }
+    if (!updateAliasesFk(db, upgradeQueries)) {
+        return false;
+    }
+    updateTextMessagesFk(upgradeQueries);
+    updateFileTransfersFk(upgradeQueries);
+    replaceOldTables(upgradeQueries);
+    return true;
+}
+
+void DbTo11::PeersToAuthors::createNewTables(QVector<RawDatabase::Query>& upgradeQueries)
+{
+    upgradeQueries += RawDatabase::Query(QStringLiteral(
+        "CREATE TABLE authors (id INTEGER PRIMARY KEY, "
+        "public_key BLOB NOT NULL UNIQUE);"));
+    upgradeQueries += RawDatabase::Query(QStringLiteral(
+        "CREATE TABLE aliases_new ("
+        "id INTEGER PRIMARY KEY, "
+        "owner INTEGER, "
+        "display_name BLOB NOT NULL, "
+        "UNIQUE(owner, display_name), "
+        "FOREIGN KEY (owner) REFERENCES authors(id));"));
+    upgradeQueries += RawDatabase::Query(QStringLiteral(
+        "CREATE TABLE text_messages_new "
+        "(id INTEGER PRIMARY KEY, "
+        "message_type CHAR(1) NOT NULL CHECK (message_type = 'T'), "
+        "sender_alias INTEGER NOT NULL, "
+        "message BLOB NOT NULL, "
+        "FOREIGN KEY (id, message_type) REFERENCES history(id, message_type), "
+        "FOREIGN KEY (sender_alias) REFERENCES aliases_new(id));"));
+    upgradeQueries += RawDatabase::Query(QStringLiteral(
+        "CREATE TABLE file_transfers_new "
+        "(id INTEGER PRIMARY KEY, "
+        "message_type CHAR(1) NOT NULL CHECK (message_type = 'F'), "
+        "sender_alias INTEGER NOT NULL, "
+        "file_restart_id BLOB NOT NULL, "
+        "file_name BLOB NOT NULL, "
+        "file_path BLOB NOT NULL, "
+        "file_hash BLOB NOT NULL, "
+        "file_size INTEGER NOT NULL, "
+        "direction INTEGER NOT NULL, "
+        "file_state INTEGER NOT NULL, "
+        "FOREIGN KEY (id, message_type) REFERENCES history(id, message_type), "
+        "FOREIGN KEY (sender_alias) REFERENCES aliases_new(id));"));
+}
+
+bool DbTo11::PeersToAuthors::populateAuthors(RawDatabase &db, QVector<RawDatabase::Query> &upgradeQueries)
```

Comment:
nit: I couldn't place why I felt weird reading these last time, but I think it's because these functions are all named like they're executing the transaction now. I think a slight naming improvement would be to name these as "appendPopulateAuthorQueries". Same for the unit tests.

Reply:
Agreed that makes more sense, renamed DbTo11 and Test10to11 functions to fit the "append.*Queries" style.